### PR TITLE
Fix undefined files handling for non-vsc editors

### PR DESCRIPTION
**Pull request by pkukielka**
## Changes

Added new client capability which is partial support for `undefined` files.
Some editors like IntelliJ does not support in-memory files, but are able to create real files on the disk if given url is a valid path.

Jetbrains PR: https://github.com/sourcegraph/jetbrains/pull/1474

## Test plan

Manual tests with Jetbrains extension and `Generate Unit Test` command.

### CODE REVIEW

**Review by mkondratek — APPROVED** [review]

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated/ClientCapabilities.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ data class ClientCapabilities(
   val progressBars: ProgressBarsEnum? = null, // Oneof: none, enabled
   val edit: EditEnum? = null, // Oneof: none, enabled
   val editWorkspace: EditWorkspaceEnum? = null, // Oneof: none, enabled
-  val untitledDocuments: UntitledDocumentsEnum? = null, // Oneof: none, enabled
```
Comment (mkondratek) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated/ClientCapabilities.kt:13]: I see the warning still appears: 
```
Cody by Sourcegraph: █ vscode.workspace.openTextDocument: unable to read non-file URI: untitled:/Users/mkondratek/runIdeProjects/jetbrains/src/test/java/MainTest.java
```

can we hide it when `onlyRealFiles`? that would reduce the noise in the logs

Comment (pkukielka) [reply agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated/ClientCapabilities.kt:13]: Fixed

File: agent/src/AgentWorkspaceDocuments.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export class AgentWorkspaceDocuments implements vscode_shim.WorkspaceDocuments {
                     'File does not exist',
                     uri.toString()
                 )
-            } else if (uri.scheme === 'file') {
+            } else if (uri.scheme === 'file' || uri.scheme === 'untitled') {
                 // Read the file content from disk if the user hasn't opened this file before.
                 const buffer = await fspromises.readFile(uri.fsPath, 'utf8')
                 document.underlying.content = buffer.toString()
```
Comment (mkondratek) [inline agent/src/AgentWorkspaceDocuments.ts:166]: This is more than changing the logging. We are modifying `document.underlying.content` now. idk if that's ok

**Review by mkondratek — APPROVED** [review]

File: agent/src/AgentWorkspaceDocuments.ts (hunk 2 above)
Comment (mkondratek) [reply agent/src/AgentWorkspaceDocuments.ts:166]: verified - it still works 😄

Comment (pkukielka) [reply agent/src/AgentWorkspaceDocuments.ts]: That change was a bit stupid on my side, it is better now :)

**Pull request merged**